tests/solver-by-substitution.cc

```diff
@@ -1035,23 +1035,27 @@ BOOST_AUTO_TEST_CASE (merge)
   vector_t error1 (c1->function ().outputSpace ()->nv ());
   vector_t error2 (c2->function ().outputSpace ()->nv ());
   vector_t error3 (c3->function ().outputSpace ()->nv ());
-  bool satisfied (solver3.isSatisfied (q, error));
-  bool found;
-  bool satisfied1 (solver3.isConstraintSatisfied (c1, q, error1, found));
-  BOOST_CHECK (found);
-  bool satisfied2 (solver3.isConstraintSatisfied (c2, q, error2, found));
-  BOOST_CHECK (found);
-  bool satisfied3 (solver3.isConstraintSatisfied (c3, q, error3, found));
-  BOOST_CHECK (found);
-  BOOST_CHECK (satisfied == (satisfied1 && satisfied2 && satisfied3));
-  size_type row = 0, nRows = error1.size ();
-  BOOST_CHECK (error1 == error.segment (row, nRows));
-  row += nRows;
-  nRows = error3.size ();
-  BOOST_CHECK (error3 == error.segment (row, nRows));
-  row += nRows;
-  nRows = error2.size ();
-  BOOST_CHECK (error2 == error.segment (row, nRows));
+  for (int i = 0; i < 2; ++i) {
+    bool satisfied (solver3.isSatisfied (q, error));
+    bool found;
+    bool satisfied1 (solver3.isConstraintSatisfied (c1, q, error1, found));
+    BOOST_CHECK (found);
+    bool satisfied2 (solver3.isConstraintSatisfied (c2, q, error2, found));
+    BOOST_CHECK (found);
+    bool satisfied3 (solver3.isConstraintSatisfied (c3, q, error3, found));
+    BOOST_CHECK (found);
+    BOOST_CHECK_EQUAL (satisfied, (satisfied1 && satisfied2 && satisfied3));
+    size_type row = 0, nRows = error1.size ();
+    BOOST_CHECK (error1 == error.segment (row, nRows));
+    row += nRows;
+    nRows = error2.size ();
+    BOOST_CHECK (error2 == error.segment (row, nRows));
+    row += nRows;
+    nRows = error3.size ();
+    BOOST_CHECK (error3 == error.segment (row, nRows));
```

Comment:
@florent-lamiraux I had to invert the check on error2 and error3 to get it to work. Is this a correct change ? Did this work before ? Note that the test was done on uninitialized values, so it may be the reason why it succeeded.

I also added a second check on a random configuration. With the neutral config, almost all the errors are 0.

Reply:
I would say that you are right. The comment line 1035 agrees with your interpretation.

Reply:
Actually, I had double free issues on the unit tests of this package on different distributions. It wasn't the case on 16.04 and 18.04 so I didn't report it.

Reply:
@nim65s Do you know the exact line ?

Reply:
I gave it another try to reproduce. And in fact, this was a `free(): invalid pointer`, after the end of the multithread test (after `*** No errors detected`) , sorry x)

I don't remember where the double free was.

Reply:
> And in fact, in this package, this was a `free(): invalid pointer`, after the end of the multithread test (after `*** No errors detected`) , sorry x)

Thanks for the notice. I keep this in mind.